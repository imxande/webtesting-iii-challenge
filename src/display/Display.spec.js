// Test away!
import React from 'react';
import Display from './Display';
import {render} from '@testing-library/react';


test('check if closed', () =>{
    const {getByTestId, rerender} = render(<Display closed/>)
    getByTestId('closed');

    rerender(<Display closed = {false} />)
    getByTestId('closed')
})

test('check if locked', () =>{
    const {getByTestId, rerender} = render(<Display locked/>)
    getByTestId('locked');

    rerender(<Display locked = {false} />)
    getByTestId('locked')
})

test('check if green red', () => {
    const className = 'led red-led';
    const locked = false;
    const display = render(<Display closed locked = {locked}
    className = {className}/>)
    expect(display.baseElement).toMatchSnapshot();
})

test('check if green green',() => {
    const className = 'led green-led';
    const locked = false;
    const closed = false;
    const display = render(<Display closed locked = {locked}
    className = {className}/>)
    expect(display.baseElement).toMatchSnapshot();
})

test('check if red red',() => {
    const className = 'led red-led';
    const locked = true;
    const closed = true;
    const display = render(<Display closed = {closed} locked = {locked}
    className = {className}/>)
    expect(display.baseElement).toMatchSnapshot();
})
