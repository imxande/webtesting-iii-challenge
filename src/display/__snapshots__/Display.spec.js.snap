// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if green green 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
        data-testid="locked"
      >
        Unlocked
      </div>
      <div
        class="led red-led"
        data-testid="closed"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;

exports[`check if green red 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
        data-testid="locked"
      >
        Unlocked
      </div>
      <div
        class="led red-led"
        data-testid="closed"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;

exports[`check if red red 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led red-led"
        data-testid="locked"
      >
        Locked
      </div>
      <div
        class="led red-led"
        data-testid="closed"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;
